### ISSUE TITLE
Reactive forms inside virtual scroll list yield buggy behaviour

### ISSUE THREAD

**Issue by fahadmajeed** [angular/angular#26799]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

Reactive forms inside a virtual scroll list, should behave as expected. It looks like virtual scroll takes over and manipulates index which yields in erratic behaviour. If we check the last item in form array as checked and then go back to top and come back again to bottom, it is marked checked again. Tried external index as well but that didn't help either.

## Expected behavior

Reactive forms especially checkboxes should behave normally, and their values should belong to 
appropriate ones only. Marking one checked should not affect others. 

## What is the motivation / use case for changing the behavior?

For heavy or huge reactive forms, virtual lists should render form controls normally.

## Environment
Angular 7 and CDK 7
<pre><code>
Angular version: 7.0.0

Browser:
- [x] Chrome (desktop) version 67.0.3396.87
- [x] Firefox version latest
- [x] Safari (desktop) version 12
- [x] IE version 11

Others:
Code sample can be seen on [stackblitz](https://stackblitz.com/edit/angular-mnd3k1) or [here](https://stackblitz.com/edit/angular-mnd3k1)

</code></pre>

**Comment by SiddAjmera** [comment]
Yeap. Seeing the same issue. I just forked the StackBlitz and printed the index as well so that it's apparent.

Two issues that I found:

1. Checking some checkboxes and then scrolling down and then scrolling up again makes the check disappear.
2. Checking a checkbox checks another checkbox as well sometimes.

Here's the [forked StackBlitz](https://stackblitz.com/edit/angular-virtual-scroll-eg?file=src%2Fapp%2Fapp.module.ts).
